### PR TITLE
Delete ClusterQueue.Lendable

### PR DESCRIPTION
**Pull request by gabesaba**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
This field is only used to populate the Cohort.Lendable field. We can populate that field directly

Related to #2502 cleanup

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: pkg/cache/clusterqueue_test.go [hunk 1]
```diff
@@ -1074,30 +1075,38 @@ func TestDominantResourceShare(t *testing.T) {
...
-					},
-				},
-				&clusterQueue{
-					Lendable: map[corev1.ResourceName]int64{
-						corev1.ResourceCPU: 2_000,
-					},
-				},
-			),
-		},
+	cache := New(utiltesting.NewFakeClient())
```
Comment (mimowo) [inline pkg/cache/clusterqueue_test.go:1078]: Why is the test changed? If the previous test case still passes maybe we could have 2 test cases (old and new) to make it clear there is no regression in coverage.

Comment (gabesaba) [reply pkg/cache/clusterqueue_test.go:1078]: deleted the underlying field - it was relying on ClusterQueue.lendable

Comment (mimowo) [reply pkg/cache/clusterqueue_test.go:1078]: ah got

**Review by mimowo — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**